screwdriver/api/api.go

```diff
@@ -33,102 +38,211 @@ type ResponseError struct {
+		return nil, err
 	}
-	defer res.Body.Close()
-	body, err := handleResponse(res)
+	var payload util.PayloadYaml
+	payload.Yaml = string(commandSpecBytes)
+	specBytes, err := json.Marshal(payload)
+	bodyBuff = bytes.NewBuffer(specBytes)
+	if err != nil {
+		return nil, err
+	}
+	return
+}
 
+func writeMultipartYaml(writer *multipart.Writer, specPath string, commandSpec *util.CommandSpec) error {
+	specBytes, err := json.Marshal(commandSpec)
+	if err != nil {
+		return fmt.Errorf("Failed to convert spec to bytes:%v", err)
+	}
+
+	yamlPart, err := writer.CreateFormField("spec")
+	if err != nil {
+		return fmt.Errorf("Failed to load spec file:%v", err)
+	}
+
+	_, err = yamlPart.Write(specBytes)
+	if err != nil {
+		return fmt.Errorf("Failed to write yaml part to the request body:%v", err)
+	}
+
+	return nil
+}
+
+func writeMultipartBin(writer *multipart.Writer, commandSpec *util.CommandSpec) error {
+	fileContentsBin, err := util.LoadByte(commandSpec.Binary.File)
+	if err != nil {
+		return fmt.Errorf("Failed to load binary file:%v", err)
+	}
+
+	fileNameBin := filepath.Base(commandSpec.Binary.File)
+	binPart, err := writer.CreateFormFile("binary", fileNameBin)
+	if err != nil {
+		return fmt.Errorf("Failed to create form of binary:%v", err)
+	}
+
+	_, err = binPart.Write(fileContentsBin)
+	if err != nil {
+		return fmt.Errorf("Failed to write binary to form:%v", err)
+	}
+
+	return nil
+}
+
+func (c client) httpRequest(httpMethod string, uri string, contentType string, body *bytes.Buffer) (*util.CommandSpec, error) {
+	responseBytes, statusCode, err := c.sendHTTPRequest(httpMethod, uri, contentType, body)
+	if err != nil {
+		return nil, fmt.Errorf("Post request failed: %v", err)
+	}
+
+	responseBytes, err = handleResponse(responseBytes, statusCode)
+	if err != nil {
+		return nil, err
+	}
+
+	responseSpec := &util.CommandSpec{}
+	err = json.Unmarshal(responseBytes, responseSpec)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse json response %v", err)
+	}
+
+	return responseSpec, err
+}
+
+func writeMultipart(specPath string, commandSpec *util.CommandSpec) (*bytes.Buffer, string, error) {
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	err := writeMultipartYaml(writer, specPath, commandSpec)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = writeMultipartBin(writer, commandSpec)
+	if err != nil {
+		return nil, "", err
+	}
+	writer.Close()
+
+	contentType := writer.FormDataContentType()
+
+	return body, contentType, nil
+}
+
+func (c client) PostCommand(specPath string, commandSpec *util.CommandSpec) (*util.CommandSpec, error) {
+	var (
+		body        *bytes.Buffer
+		contentType string
+		err         error
+	)
+	switch commandSpec.Format {
+	case "binary":
+		body, contentType, err = writeMultipart(specPath, commandSpec)
+	case "habitat", "docker":
+		body, err = specToPayloadBuf(commandSpec)
+		contentType = "application/json"
+	}
```

Comment:
It will not happen but is it better to check unknown words by `default`?

Reply:
Thank you! I've fixed.